lib/capybara/selector.rb

```diff
@@ -380,15 +381,21 @@
   end
 
   filter(:selected) do |node, selected|
-    actual = node.all(:xpath, './/option', visible: false).select { |option| option.selected? }.map { |option| option.text(:all) }
-    [selected].flatten.sort == actual.sort
+    actual = node.all(:xpath, './/option', visible: false).select(&:selected?).map { |option| option.text(:all) }
+    Array(selected).sort == actual.sort
+  end
+
+  filter(:with_selected) do |node, selected|
+    actual = node.all(:xpath, './/option', visible: false).select(&:selected?).map { |option| option.text(:all) }
```

Comment:
Im not too happy about this line repeating in :select and :with_selected filters, though I didn't want to extract this to method/lambda to not break the very clear structure of this file.